demo/clusterctl-examples/README.md

```diff
@@ -0,0 +1,150 @@
+# Table Of Contents
+
+- [Overview](#overview)
+  - [OS Image](#os_image)
+  - [Network Configurations](#network_configurations)
+  - [Network Access](#network_access)
+  - [User Accounts](#user_accounts)
+- [Prerequsites](#prerequisites)
+- [How-To](#how_to)
+
+# Overview <a name="overview" />
+
+This directory contains sample [clusterctl templates][clusterctl-templates] for
+both [cri-o][cri-o] and [rke2][rke2] clusters. The boostrapping and
+configuration of these clusters will be largely depended on the following:
+
+## OS Image <a name="os_image" />
+
+The type of OS image (i.e. SUSE-based, Ubuntu-based, etc) and what's on that
+image (i.e. packages pre-installed, package availability, networking setup,
+security setup, user accounts setup, etc) will determine what the cluster
+bootstrapping commands will look like.
+
+## Network Configurations <a name="network_configurations" />
+
+The network configuration is very much environment-specific. *The NIC names
+for the baremetal nodes must be consistent as they are sharing the same
+Metal3DataTemplate*. For example, we can't have a NIC in one node named
+`eno1f0` while the other node have it as `eth0`.
+
+Furthermore, the configuration of IP pools, gateways, DNS, and routing table
+requires indepth knowledge of infrastructure networking setup.
+
+## Network Access <a name="network_access" />
+
+Whether the baremetal provisioning network have public network egress or not
+(i.e. air-gapped environment) will also influence the cluster bootstrapping
+commands. For an air-gapped environment, any public endpoints will need to be
+updated to their respective internal endpoints.
+
+## User Accounts <a name="user_accounts" />
+
+The bootstrapping commands may also depended on certain user accounts (for
+management and monitoring purposes as an example).
```

Comment:
At the minimum, we should have a user that can SSH into the node in case there are failures which required local troubleshooting. If we need to install addition monitoring software on the node (i.e. auditing, log streaming, etc), addition user accounts may be needed if we don't want to run those as root user.